### PR TITLE
Repo reorganization and rename

### PR DESCRIPTION
**Pull request by ympek**
Split repo directories to CLI, daemon, core and platforms

### CODE REVIEW

File: core/husarnet.cmake [hunk 1]
```diff
@@ -81,16 +81,22 @@ endif()
...
 include_directories(${CMAKE_CURRENT_LIST_DIR})
 
 file(GLOB core_SRC "${CMAKE_CURRENT_LIST_DIR}/*.cpp")
 list(APPEND husarnet_core_SRC ${core_SRC})
 
+# "Aliasing" (copying) core folder as "husarnet"
+# So includes like "husarnet/something.h" do work too
+set(TEMP_INCLUDE_DIR ${CMAKE_BINARY_DIR}/tempIncludes)
+file(MAKE_DIRECTORY ${TEMP_INCLUDE_DIR})
+file(COPY ../../core/ DESTINATION ${TEMP_INCLUDE_DIR}/husarnet/ FILES_MATCHING PATTERN "*.h")
```
Comment (pidpawel) [inline core/husarnet.cmake:93]: You can add `*.hpp` for future compatibility too

File: core/husarnet.cmake [hunk 2]
```diff
@@ -81,16 +81,22 @@ endif()
...
+
 # Join all of the above
 add_library(husarnet_core STATIC ${husarnet_core_SRC})
 
+target_include_directories(husarnet_core PUBLIC ${TEMP_INCLUDE_DIR})
```
Comment (pidpawel) [inline core/husarnet.cmake:98]: Kinda feels like that line belongs to that section above a little bit more

**Comment by pidpawel** [comment]
1) `utils/deploy.sh` should also be changed to reflect the `deploy/static` -> `deploy` migration
2) in `platforms/unix` the `-misc` suffix seems redundant now
Other than those couple of things - I like it!

File: core/husarnet.cmake (hunk 1 above)
Comment (ympek) [reply core/husarnet.cmake:93]: Great idea thanks

File: core/husarnet.cmake (hunk 2 above)
Comment (ympek) [reply core/husarnet.cmake:98]: I agree, but as per CMake docs: 
target_include_directories
Specifies include directories to use when compiling a given target. The named <target> must have been created by a command such as add_library (...)

implies `add_library` must come before any `target_include_directories` and I can confirm that this indeed does not work after moving up

**Review by pidpawel — APPROVED** [review]
LGTM!

**Pull request merged**